tests/phpunit/tests/rest-api/rest-global-styles-controller.php

```diff
@@ -132,30 +149,85 @@ public function test_get_theme_item_permission_check() {
 		$this->assertErrorResponse( 'rest_cannot_manage_global_styles', $response, 403 );
 	}
 
-
 	/**
 	 * @covers WP_REST_Global_Styles_Controller::get_theme_item
 	 * @ticket 54516
 	 */
-	public function test_get_theme_item_invalid() {
+	public function test_get_theme_item_not_found() {
 		wp_set_current_user( self::$admin_id );
-		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/invalid' );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/doesnotexist' );
 		$response = rest_get_server()->dispatch( $request );
 		$this->assertErrorResponse( 'rest_theme_not_found', $response, 404 );
 	}
 
 	/**
+	 * @dataProvider data_get_theme_item_invalid_theme_dirname
 	 * @covers WP_REST_Global_Styles_Controller::get_theme_item
+	 * @ticket 54516
 	 */
-	public function test_get_theme_item() {
+	public function test_get_theme_item_invalid_theme_dirname( $theme_dirname ) {
 		wp_set_current_user( self::$admin_id );
-		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/tt1-blocks' );
+		switch_theme( $theme_dirname );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/' . $theme_dirname );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertErrorResponse( 'rest_no_route', $response, 404 );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_get_theme_item_invalid_theme_dirname() {
+		return array(
+			'with |'                 => array( 'my|theme' ),
+			'with +'                 => array( 'my+theme' ),
+			'with {}'                => array( 'my{theme}' ),
+			'with #'                 => array( 'my#theme' ),
+			'with !'                 => array( 'my!theme' ),
+			'multiple invalid chars' => array( 'mytheme-[_(+@)]#! v4.0' ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_get_theme_item
+	 * @covers WP_REST_Global_Styles_Controller::get_theme_item
+	 * @ticket 54516
+	 */
+	public function test_get_theme_item( $theme ) {
+		wp_set_current_user( self::$admin_id );
+		switch_theme( $theme );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/themes/' . $theme );
 		$response = rest_get_server()->dispatch( $request );
 		$data     = $response->get_data();
 		unset( $data['_links'] );
```

Comment:
Why do we `unset( ['_links'] )`?

Reply:
No idea. But to minimize changes, I left it for this PR.